04_quicksort/rust/src/main.rs

```diff
@@ -0,0 +1,140 @@
+fn rec_sum(list: &[usize]) -> usize {
+    match list.get(0) {
+        Some(x) => x + rec_sum(&list[1..]),
+        None => 0,
+    }
+}
+
+fn rec_count<T>(list: &[T]) -> usize {
+    match list.get(0) {
+        Some(_) => 1 + rec_count(&list[1..]),
+        None => 0,
+    }
+}
+
+fn maximum<T: Ord>(list: &[T]) -> Option<&T> {
+    match list.get(0) {
+        Some(x) => match maximum(&list[1..]) {
+            Some(max) => {
+                if x > max {
+                    Some(x)
+                } else {
+                    Some(max)
+                }
+            }
+            None => Some(x),
+        },
+        None => None,
+    }
+}
+
+fn binary_search<T: Ord>(list: &[T], item: T) -> Option<usize> {
```

Comment:
Could you get rid of the binary search code?

Reply:
@egonSchiele sorry about that. Must have been a held over from the binary search. Removed now.